### ISSUE TITLE
Cannot set default font for SQL language

### ISSUE THREAD

**Issue by mdell-seradex** [notepad-plus-plus/notepad-plus-plus#14021]
### Description of the Issue
I want to change the default font used for SQL files where the language is identified (manual or auto) as SQL.
Other languages have a DEFAULT style.  It is my understanding that this is where that is set.
In my installation, SQL does not have this.  I tried changing the font for a few styles, but none had any effect.

### Steps to Reproduce the Issue
1. In the `Settings` menu select `Style Configurator` to open it.
2. Select the `SQL` language.
3. See that there is no `DEFAULT` style.

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/40300659/c065b241-5db6-4e38-b063-a116ab1355d9)

### Expected Behavior
I expected to have a `DEFAULT` style where I could set the font.

### Actual Behavior
I could not do that.

### Debug Information
Notepad++ v8.5.6   (32-bit)
Build time : Aug 15 2023 - 15:27:06
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Command Line : "C:\Program Files (x86)\Notepad++\change.log" 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit)
OS Version : 22H2
OS Build : 19045.3208
Current ANSI codepage : 1252
Plugins : 
    DSpellCheck (1.4.24)
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
    NPPJSONViewer (1.40)
    XMLTools (3.1.0.1)

**Comment by victorel-petrovich** [comment]
> In my installation, SQL does not have this. [default style]

Same in my installation...

> I tried changing the font for a few styles, but none had any effect.

I _can_ change font (name and size) for String, for example. Didn't check others as I don't have a proper SQL file. 

Sidenote: trying to configure styles for Python or PHP language, they have Default style, but changing it has no effect (other than changing height of lines). 
So I wonder whether it was on purpose that SQL was left without Default style, so that you have to customize style for each element. 

Notepad++ v8.5.6   (32-bit)
Build time : Aug 15 2023 - 15:27:06
Path : C:\Users\Vic\Downloads\npp.8.5.6.portable\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 7 Ultimate (32-bit)
OS Build : 7601.24546
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
    PythonScript (2)

**Comment by mdell-seradex** [comment]
Actually, I was not completely correct.
When I change the Font Name, the line spacing in the document changes, but not the actual font.
For instance, if I select "Times New Roman" as the font, nothing is displayed in the "Times New Roman" font.
Same is true for other fonts like "Algerian".

Am I trying to do this the wrong way?  Does Notepad++ not actually have that ability?

**Comment by mpheath** [comment]
> When I change the Font Name, the line spacing in the document changes, but not the actual font.
> For instance, if I select "Times New Roman" as the font, nothing is displayed in the "Times New Roman" font.

Depends on the style being changed to what effect will occur.
If you set one style to a larger font, then the lines may alter in size to accommodate the change.
Changing the font for **SQL** -> **KEYWORD**, I see the font change for keywords like `SELECT`, `FROM`, ...
Each style can be independently set a font, size, ...

If you want a more global font change, then you could try **Global Styles** -> **Default Style**.
Just be careful not to mess things up too much. Small change and reset back if needed ASAP.
As to the title, there is no actual "default font name for SQL language" as each style can be customized. I guess you may need to experience it yourself and figure out of what suits your taste.

**Comment by victorel-petrovich** [comment]
I agree that it is confusing to have, for many of the languages, a style called "Default", but it doesn't do what this name would imply: to be the style that any of the other elements and styles fall back on, if those elements/styles do not specify their own. 

But this would have to be complained at https://www.scintilla.org/Lexilla.html , as Npp just uses the lexers from there, if I understand correctly.

**Comment by mdell-seradex** [comment]
I get you.
I was just wanting to change the font name that is used for SQL Files, but I guess the ability does not exist.
To me it makes sense to be able to specify a different font name per language, if that is what you want.
In most cases, I prefer a font like "Microsoft Sans Serif", "Calibri", or "Segoe UI", but for SQL, I use "Consolas".
I just want to set it once and not have to keep changing the font back and forth.

So then, this seems to have changed into an enhancement request.

**Comment by victorel-petrovich** [comment]
> To me it makes sense to be able to specify a different font name per language, if that is what you want.

👍 . Not only font name , but also other font attributes (size, etc, as usual). 
If implemented, that should get the name DEFAULT, not ... whitespace, or whatever is style 0 in diff. languages.

**Comment by mdell-seradex** [comment]
It actually seems strange to set a different font name for part of it, if it doesn't change that actuall displayed font for that.
I would not be adverse to removing that, or only enabling it for languages that can apply that kind of change.
It doesn't feel useful to have a font name change affect only the line spacing.
